In [1]:
import pandas as pd
from scipy import stats

df = pd.read_csv('../data/clean_data.csv')

## Hypothesis Testing — Validating EDA Findings
Bivariate analysis suggested Area and Location matter most for Price. Here we formally test each relationship using appropriate statistical tests (ANOVA for categorical predictors, Pearson correlation for numeric, Chi-square for categorical-categorical) at α = 0.05.

In [2]:
# --- Hypothesis 1: Does Location affect Price? ---
# H0: mean Price is the same across all Locations
# H1: at least one Location has a different mean Price
# One-way ANOVA: compares means across 3+ groups

groups = [df[df['Location']==loc]['Price'] for loc in df['Location'].unique()]
f_stat, p_val = stats.f_oneway(*groups)
print(f"F-statistic: {f_stat:.2f}, p-value: {p_val:.6f}")
print("Significant → reject H0" if p_val < 0.05 else "Not significant → fail to reject H0")

F-statistic: 58.68, p-value: 0.000000
Significant → reject H0


In [3]:
# --- Hypothesis 2: Does Property_Type affect Price? ---
# Same test, different grouping variable
groups2 = [df[df['Property_Type']==pt]['Price'] for pt in df['Property_Type'].unique()]
f_stat2, p_val2 = stats.f_oneway(*groups2)
print(f"F-statistic: {f_stat2:.2f}, p-value: {p_val2:.6f}")
print("Significant → reject H0" if p_val2 < 0.05 else "Not significant → fail to reject H0")

F-statistic: 2.80, p-value: 0.062168
Not significant → fail to reject H0


In [4]:
# --- Hypothesis 3: Is the Area-Price correlation statistically real? ---
# H0: true correlation is 0 (no relationship)
r, p_val3 = stats.pearsonr(df['Area'], df['Price'])
print(f"r = {r:.3f}, p-value: {p_val3:.2e}")
print("Significant → reject H0" if p_val3 < 0.05 else "Not significant → fail to reject H0")

r = 0.796, p-value: 5.11e-67
Significant → reject H0


In [5]:
# --- Hypothesis 4: Are Location and Property_Type associated? ---
# H0: the two categorical variables are independent
contingency = pd.crosstab(df['Location'], df['Property_Type'])
chi2, p_val4, dof, expected = stats.chi2_contingency(contingency)
print(f"chi2 = {chi2:.2f}, p-value: {p_val4:.4f}")
print("Significant → reject H0" if p_val4 < 0.05 else "Not significant → fail to reject H0")

chi2 = 0.45, p-value: 0.9779
Not significant → fail to reject H0


## Key Findings
- Location significantly affects Price (ANOVA, p < 0.001) — confirms EDA
- Property_Type does NOT significantly affect Price (ANOVA, p = 0.062) — EDA's visual impression was misleading; not statistically supported
- Area-Price correlation is highly significant (p < 0.001) — strongly confirms EDA
- Location and Property_Type are independent (Chi-square, p = 0.978)